### ISSUE TITLE
Some chunks on minimap become black

### ISSUE THREAD

**Issue by LuciusV** [FTBTeam/FTB-Chunks#126]
Mod versions: 
ftb-chunks: 1604.2.2.62
ftb-gui-library : 1605.2.1.39
World seed: -5594663186523761295

Some explored areas on map can become black again
Every time it happens it happens for same chunks, see below
![Screenshot 2021-03-16 134511](https://user-images.githubusercontent.com/11925386/111261910-2de59300-865e-11eb-9099-4039c4ae1f47.png)

**Comment by LatvianModder** [comment]
Thats super weird. What modpack is that and does it only reset after restarting game/world, or does it also happen if you just move away?

**Comment by LuciusV** [comment]
Its my own kitchen sink modpack. Do you need logs and modlist? About reset, I will test it and report exact conditions when it happens.

**Comment by LuciusV** [comment]
So I tested and found that it happens after restarting world

**Comment by Kipex** [comment]
I'm having what looks like the same issue. Bunch of chunks are black when I open up the game and join our private ATM6 server. This has been happening in at least 1.5.3b and 1.5.5 releases of ATM6, I think 1.5.1 as well which is where we started playing.

I can fly over the chunks to sort of paint them back, but they'll be black again once I restart my game.

![image](https://user-images.githubusercontent.com/25510049/111908777-37f30180-8a63-11eb-837a-d5da3598d35f.png)

**Comment by G-Stav** [comment]
I'm having similar issues on a custom pack, but it appears to only affect chunks near spawn as far as I know.
![image](https://user-images.githubusercontent.com/18726116/125293052-f4c2e400-e31a-11eb-9cce-69790b1c5568.png)

**Comment by Spudstak** [comment]
Also experiencing the same issue as @G-Stav

**Comment by mcpower** [comment]
Encountering the same problem.

- FTB Academy 1.16 v1.1.0, no modifications downloaded directly using MultiMC
- Spawn chunks appear black on overworld (after doing `/school exit` to exit out of the initial school).
- Consistently reproduces (all six attempts exhibited the bug).
- Does not affect the "school" world.
- Does not affect chunks outside initial chunk generation (see screenshots of "render distance of 2 chunks").

Minecraft details:

- Run with MultiMC nightly (at https://github.com/MultiMC/MultiMC5/commit/0c466bc530dc7e6da12d678d01677dfb23bdda92)
- Windows 10 20H2
- AdoptOpenJDK 15.0.2+7 running with `-XX:+UnlockExperimentalVMOptions -XX:+UseZGC`. Also reproduced with AdoptOpenJDK 8.0.292+1 running with `-XX:+UseG1GC -Dsun.rmi.dgc.server.gcInterval=2147483646 -XX:+UnlockExperimentalVMOptions -XX:G1NewSizePercent=20 -XX:G1ReservePercent=20 -XX:MaxGCPauseMillis=50 -XX:G1HeapRegionSize=32M`.

<details>
<summary>Initial MultiMC boot log</summary>

```
MultiMC version: 0.6.12-develop-2241


Launched instance in online mode
Authentication server is online

authserver.mojang.com resolves to:
    [redacted IP]

session.minecraft.net resolves to:
    [redacted IP, redacted IP, redacted IP]

textures.minecraft.net resolves to:
    [redacted IP, redacted IP, redacted IP, redacted IP]

api.mojang.com resolves to:
    [redacted IP]


Minecraft folder is:
D:/Games/MultiMC/instances/FTB Academy 1.16 v1.1.0/minecraft


Java path is:
C:/Program Files/AdoptOpenJDK/jdk15/bin/javaw.exe


Checking Java version...
Java is version 15.0.2, using 64-bit architecture, from AdoptOpenJDK.


Main Class:
  io.github.zekerzhayard.forgewrapper.installer.Main

Native path:
  D:/Games/MultiMC/instances/FTB Academy 1.16 v1.1.0/natives

Traits:
traits XR:Initial
traits FirstThreadOnMacOS

Libraries:
  D:/Games/MultiMC/libraries/org/lwjgl/lwjgl-glfw/3.2.2/lwjgl-glfw-3.2.2.jar
  D:/Games/MultiMC/libraries/org/lwjgl/lwjgl-jemalloc/3.2.2/lwjgl-jemalloc-3.2.2.jar
  D:/Games/MultiMC/libraries/org/lwjgl/lwjgl-openal/3.2.2/lwjgl-openal-3.2.2.jar
  D:/Games/MultiMC/libraries/org/lwjgl/lwjgl-opengl/3.2.2/lwjgl-opengl-3.2.2.jar
  D:/Games/MultiMC/libraries/org/lwjgl/lwjgl-stb/3.2.2/lwjgl-stb-3.2.2.jar
  D:/Games/MultiMC/libraries/org/lwjgl/lwjgl-tinyfd/3.2.2/lwjgl-tinyfd-3.2.2.jar
  D:/Games/MultiMC/libraries/org/lwjgl/lwjgl/3.2.2/lwjgl-3.2.2.jar
  D:/Games/MultiMC/libraries/com/mojang/patchy/1.2.3/patchy-1.2.3.jar
  D:/Games/MultiMC/libraries/oshi-project/oshi-core/1.1/oshi-core-1.1.jar
  D:/Games/MultiMC/libraries/net/java/dev/jna/jna/4.4.0/jna-4.4.0.jar
  D:/Games/MultiMC/libraries/net/java/dev/jna/platform/3.4.0/platform-3.4.0.jar
  D:/Games/MultiMC/libraries/com/ibm/icu/icu4j/66.1/icu4j-66.1.jar
  D:/Games/MultiMC/libraries/com/mojang/javabridge/1.0.22/javabridge-1.0.22.jar
  D:/Games/MultiMC/libraries/net/sf/jopt-simple/jopt-simple/5.0.4/jopt-simple-5.0.4.jar
  D:/Games/MultiMC/libraries/io/netty/netty-all/4.1.25.Final/netty-all-4.1.25.Final.jar
  D:/Games/MultiMC/libraries/com/google/guava/guava/21.0/guava-21.0.jar
  D:/Games/MultiMC/libraries/org/apache/commons/commons-lang3/3.5/commons-lang3-3.5.jar
  D:/Games/MultiMC/libraries/commons-io/commons-io/2.5/commons-io-2.5.jar
  D:/Games/MultiMC/libraries/commons-codec/commons-codec/1.10/commons-codec-1.10.jar
  D:/Games/MultiMC/libraries/com/mojang/brigadier/1.0.17/brigadier-1.0.17.jar
  D:/Games/MultiMC/libraries/com/mojang/datafixerupper/4.0.26/datafixerupper-4.0.26.jar
  D:/Games/MultiMC/libraries/com/google/code/gson/gson/2.8.0/gson-2.8.0.jar
  D:/Games/MultiMC/libraries/com/mojang/authlib/2.1.28/authlib-2.1.28.jar
  D:/Games/MultiMC/libraries/org/apache/commons/commons-compress/1.8.1/commons-compress-1.8.1.jar
  D:/Games/MultiMC/libraries/org/apache/httpcomponents/httpclient/4.3.3/httpclient-4.3.3.jar
  D:/Games/MultiMC/libraries/commons-logging/commons-logging/1.1.3/commons-logging-1.1.3.jar
  D:/Games/MultiMC/libraries/org/apache/httpcomponents/httpcore/4.3.2/httpcore-4.3.2.jar
  D:/Games/MultiMC/libraries/it/unimi/dsi/fastutil/8.2.1/fastutil-8.2.1.jar
  D:/Games/MultiMC/libraries/org/apache/logging/log4j/log4j-api/2.11.2/log4j-api-2.11.2.jar
  D:/Games/MultiMC/libraries/org/apache/logging/log4j/log4j-core/2.11.2/log4j-core-2.11.2.jar
  D:/Games/MultiMC/libraries/com/mojang/text2speech/1.11.3/text2speech-1.11.3.jar
  D:/Games/MultiMC/libraries/io/github/zekerzhayard/ForgeWrapper/1.4.2/ForgeWrapper-1.4.2.jar
  D:/Games/MultiMC/libraries/net/minecraftforge/forge/1.16.5-36.1.4/forge-1.16.5-36.1.4-launcher.jar
  D:/Games/MultiMC/libraries/org/ow2/asm/asm/9.0/asm-9.0.jar
  D:/Games/MultiMC/libraries/org/ow2/asm/asm-commons/9.0/asm-commons-9.0.jar
  D:/Games/MultiMC/libraries/org/ow2/asm/asm-tree/9.0/asm-tree-9.0.jar
  D:/Games/MultiMC/libraries/org/ow2/asm/asm-util/9.0/asm-util-9.0.jar
  D:/Games/MultiMC/libraries/org/ow2/asm/asm-analysis/9.0/asm-analysis-9.0.jar
  D:/Games/MultiMC/libraries/cpw/mods/modlauncher/8.0.9/modlauncher-8.0.9.jar
  D:/Games/MultiMC/libraries/cpw/mods/grossjava9hacks/1.3.3/grossjava9hacks-1.3.3.jar
  D:/Games/MultiMC/libraries/net/minecraftforge/accesstransformers/3.0.1/accesstransformers-3.0.1.jar
  D:/Games/MultiMC/libraries/org/antlr/antlr4-runtime/4.9.1/antlr4-runtime-4.9.1.jar
  D:/Games/MultiMC/libraries/net/minecraftforge/eventbus/4.0.0/eventbus-4.0.0.jar
  D:/Games/MultiMC/libraries/net/minecraftforge/forgespi/3.2.0/forgespi-3.2.0.jar
  D:/Games/MultiMC/libraries/net/minecraftforge/coremods/4.0.6/coremods-4.0.6.jar
  D:/Games/MultiMC/libraries/net/minecraftforge/unsafe/0.2.0/unsafe-0.2.0.jar
  D:/Games/MultiMC/libraries/com/electronwill/night-config/core/3.6.3/core-3.6.3.jar
  D:/Games/MultiMC/libraries/com/electronwill/night-config/toml/3.6.3/toml-3.6.3.jar
  D:/Games/MultiMC/libraries/org/jline/jline/3.12.1/jline-3.12.1.jar
  D:/Games/MultiMC/libraries/org/apache/maven/maven-artifact/3.6.3/maven-artifact-3.6.3.jar
  D:/Games/MultiMC/libraries/net/jodah/typetools/0.8.3/typetools-0.8.3.jar
  D:/Games/MultiMC/libraries/net/minecrell/terminalconsoleappender/1.2.0/terminalconsoleappender-1.2.0.jar
  D:/Games/MultiMC/libraries/org/spongepowered/mixin/0.8.2/mixin-0.8.2.jar
  D:/Games/MultiMC/libraries/net/minecraftforge/nashorn-core-compat/15.1.1.1/nashorn-core-compat-15.1.1.1.jar
  D:/Games/MultiMC/libraries/com/mojang/minecraft/1.16.5/minecraft-1.16.5-client.jar

Native libraries:
  D:/Games/MultiMC/libraries/org/lwjgl/lwjgl-glfw/3.2.2/lwjgl-glfw-3.2.2-natives-windows.jar
  D:/Games/MultiMC/libraries/org/lwjgl/lwjgl-jemalloc/3.2.2/lwjgl-jemalloc-3.2.2-natives-windows.jar
  D:/Games/MultiMC/libraries/org/lwjgl/lwjgl-openal/3.2.2/lwjgl-openal-3.2.2-natives-windows.jar
  D:/Games/MultiMC/libraries/org/lwjgl/lwjgl-opengl/3.2.2/lwjgl-opengl-3.2.2-natives-windows.jar
  D:/Games/MultiMC/libraries/org/lwjgl/lwjgl-stb/3.2.2/lwjgl-stb-3.2.2-natives-windows.jar
  D:/Games/MultiMC/libraries/org/lwjgl/lwjgl-tinyfd/3.2.2/lwjgl-tinyfd-3.2.2-natives-windows.jar
  D:/Games/MultiMC/libraries/org/lwjgl/lwjgl/3.2.2/lwjgl-3.2.2-natives-windows.jar
  D:/Games/MultiMC/libraries/com/mojang/text2speech/1.11.3/text2speech-1.11.3-natives-windows.jar

Mods:
  [✔️] absentbydesign-1.16.5-1.5.0
  [✔️] appleskin-forge-mc1.16.x-2.0.0
  [✔️] architectury-1.19.27-forge
  [✔️] astralsorcery-1.16-1.16.5-1.13.12
  [✔️] AutoRegLib-1.6-49
  [✔️] BetterAdvancements-1.16.5-0.1.1.109
  [✔️] BetterCaves-Forge-1.16.4-1.1.2
  [✔️] betterendforge-1.16.4-1.6.0
  [✔️] BetterMineshafts-Forge-1.16.4-2.0.4
  [✔️] Bookshelf-1.16.5-10.1.11
  [✔️] Botania-1.16.5-418
  [✔️] BotanyPots-1.16.5-7.0.17
  [✔️] BotanyTrees-1.16.5-3.0.5
  [✔️] buildinggadgets-1.16.5-3.8.0
  [✔️] bwncr-1.16.4-3.9.16
  [✔️] byg-1.2.9
  [✔️] catalogue-1.2.1-1.16.5
  [✔️] charm-forge-1.16.5-2.3.2
  [✔️] chocolate-1.3.0-1.16.4
  [✔️] cloth-config-4.11.26-forge
  [✔️] Clumps-6.0.0.25
  [✔️] CodeChickenLib-1.16.5-4.0.2.427-universal
  [✔️] cofh_core-1.16.4-1.2.1
  [✔️] Compressium-1.16.5-1.2.2
  [✔️] Controlling-7.0.0.23
  [✔️] corpse-1.16.5-1.0.5
  [✔️] CTM-MC1.16.1-1.1.2.6
  [✔️] curios-forge-1.16.5-4.0.5.2
  [✔️] Cyclic-1.16.5-1.3.2
  [✔️] DamageTilt-1.16-forge-0.1.1
  [✔️] dankstorage-3.16
  [✔️] DarkUtilities-1.16.5-8.0.10
  [✔️] DefaultOptions_1.16.5-12.2.1
  [✔️] DrawersTooltip-1.16.2-2.1.0
  [✔️] DynamicSurroundings-1.16.4-4.0.4.2
  [✔️] dynviewdist-2.0
  [✔️] elevatorid-1.16.5-1.7.13
  [✔️] emojiful-1.16.4-2.1.4
  [✔️] enchantedbookredesign-3.3-1.16.4
  [✔️] EnchantmentDescriptions-1.16.5-7.0.12
  [✔️] EnderStorage-1.16.5-2.8.0.168-universal
  [✔️] ensorcellation-1.16.4-1.2.0
  [✔️] ExtraStorage-1.16.5-1.4.1
  [✔️] extremeSoundMuffler-3.13-1.16.5
  [✔️] fastbenchminusreplacement-1.4
  [✔️] FasterLadderClimbing-1.16.4-0.2.1
  [✔️] fastfurnaceminusreplacement-1.1-1.16.3
  [✔️] FastLeafDecay-v25
  [✔️] ferritecore-2.0.5-forge
  [✔️] findme-1.16.3-2.2.0.0
  [✔️] ftbauxilium-forge-1605.1.1-build.14-forge
  [✔️] ftb-backups-2.1.1.6
  [✔️] ftb-banners-1605.1.3-build.23
  [✔️] ftb-chunks-forge-1605.3.1-build.24
  [✔️] ftb-essentials-1605.1.4-build.4
  [✔️] ftb-janitor-1605.1.2-build.3
  [✔️] ftb-library-forge-1605.3.1-build.48
  [✔️] ftb-quests-forge-1605.3.3-build.39
  [✔️] ftb-ranks-1605.1.4-build.5-forge
  [✔️] FTB-Schools-1605.0.2-dev.16
  [✔️] ftb-teams-forge-1605.2.1-build.17
  [✔️] ftb-ultimine-forge-1605.2.2-build.4
  [✔️] GameStages-1.16.5-7.1.6
  [✔️] glassential-forge-1.16.5-1.1.6
  [✔️] GooglyEyes-1.16.5-10.4.0
  [📁] hats (folder)
  [✔️] Hats-1.16.5-10.3.1
  [✔️] iChunUtil-1.16.5-10.4.1
  [✔️] industrial-foregoing-1.16.5-3.2.14-6
  [✔️] ironchest-1.16.4-11.2.10
  [✔️] item-filters-forge-1605.2.5-build.9
  [✔️] jei-1.16.5-7.7.1.110
  [✔️] jeiintegration_1.16.5-7.0.1.15
  [✔️] jei-professions-1.0.0-1.16.4
  [✔️] justenoughbeacons-3.1-1.16.3
  [✔️] JustEnoughResources-1.16.5-0.12.1.121
  [✔️] KleeSlabs_1.16.5-9.2.1
  [✔️] Krate-1.16.5-0.1.4
  [✔️] kubejs-forge-1605.3.11-build.31
  [✔️] kubejs-thermal-1605.1.1.6
  [✔️] kubejs-ui-1605.1.4-build.10-forge
  [✔️] light-overlay-5.8.1
  [✔️] lofirecordstomineto-1.1.0
  [✔️] Lollipop-1.16.5-3.2.9
  [✔️] mcjtylib-1.16-5.0.21
  [✔️] minetogether-1.16.5-5.0.0Alpha6-forge
  [✔️] mininggadgets-1.7.5
  [✔️] modnametooltip_1.16.2-1.15.0
  [✔️] moredragoneggs-1.4
  [✔️] Morpheus-1.16.5-4.2.70
  [✔️] MouseTweaks-2.14-mc1.16.2
  [✔️] NaturesCompass-1.16.5-1.9.1-forge
  [✔️] notenoughcrashes-3.2.0-forge
  [✔️] observerlib-1.16.5-1.5.3
  [✔️] PackingTape-1.16.3-0.10.0
  [✔️] Patchouli-1.16.4-53.1
  [✔️] pipez-1.16.5-1.2.7
  [✔️] pitchperfect-1.0.2
  [✔️] Placebo-1.16.4-4.5.0
  [✔️] Powah-1.16.5-2.3.16
  [✔️] QuarkOddities-1.16.3
  [✔️] Quark-r2.4-314
  [✔️] quartz-chests-1603.1.0.8
  [✔️] ReAuth-1.16-Forge-3.9.3
  [✔️] recipebuffers-2.1
  [✔️] refinedstorage-1.9.13
  [✔️] refinedstorageaddons-0.7.2
  [✔️] rftoolsbase-1.16-2.0.11
  [✔️] rftoolsbuilder-1.16-3.1.0
  [✔️] rftoolspower-1.16-3.0.9
  [✔️] rftoolsstorage-1.16-2.0.12
  [✔️] rftoolsutility-1.16-3.1.0
  [✔️] rhino-1605.1.1-build.21
  [✔️] rsrequestify-1.16.3-2.0.1
  [✔️] Runelic-1.16.5-7.0.1
  [✔️] serverconfigupdater-1.3
  [✔️] shutupexperimentalsettings-1.0.3
  [✔️] SimpleDiscordRichPresence-1.16.4-1.3.5
  [✔️] simplybackpacks-1.16.3-1.4.13
  [✔️] SimplyJetpacks2-1.16.5-2.6.7.1
  [✔️] simplylight-1.16.4-1.1.3
  [✔️] Statues-1.16.5-0.1.6.9
  [✔️] StorageDrawers-1.16.3-8.2.2
  [✔️] StructureExpansion-1.16.5-1.0.7
  [✔️] supermartijn642configlib-1.0.8-mc1.16
  [✔️] supermartijn642corelib-1.0.9-mc1.16.5
  [✔️] swingthroughgrass-1.16.4-1.5.3
  [✔️] tanknull-2.3-1.16.4
  [✔️] tesseract-1.0.22-mc1.16.5
  [✔️] theoneprobe-1.16-3.1.4
  [✔️] thermal_cultivation-1.16.4-1.2.0
  [✔️] thermal_expansion-1.16.4-1.2.0
  [✔️] thermal_foundation-1.16.4-1.2.0
  [✔️] thermal_innovation-1.16.4-1.2.0
  [✔️] thermal_locomotion-1.16.4-1.2.0
  [✔️] Tiny-Coal-1.16.4-1.0.0
  [✔️] Tips-1.16.5-4.0.7
  [✔️] TipTheScales-1.16.5-3.0.0.15
  [✔️] titanium-1.16.5-3.2.8.4-10
  [✔️] Toast-Control-1.16.4-4.3.1
  [✔️] towers_of_the_wild-1.16.4-2.0.1
  [✔️] TrashSlot_1.16.3-12.2.1
  [✔️] upstream-1.0
  [✔️] Waystones_1.16.5-7.6.2
  [✔️] YungsApi-1.16.4-Forge-6

Params:
  --username  --version MultiMC5 --gameDir D:/Games/MultiMC/instances/FTB Academy 1.16 v1.1.0/minecraft --assetsDir D:/Games/MultiMC/assets --assetIndex 1.16 --uuid  --accessToken  --userType  --versionType release --launchTarget fmlclient --fml.forgeVersion 36.1.4 --fml.mcVersion 1.16.5 --fml.forgeGroup net.minecraftforge --fml.mcpVersion 20210115.111550

Window size: max (if available)

Java Arguments:
[-XX:+UnlockExperimentalVMOptions, -XX:+UseZGC, -XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump, -Xms1024m, -Xmx6144m, -Duser.language=en]


Minecraft process ID: 9772


Using onesix launcher.
```

</details>


<details>
<summary>Screenshot when playing (created world + entered overworld) with a render distance of 12 chunks</summary>

![image](https://user-images.githubusercontent.com/6015058/126865740-4cc1b5b6-08c9-4e2a-b3d3-5048c4a56f28.png)

</details>

<details>
<summary>Screenshots when playing (created world + entered overworld) with a render distance of 2 chunks</summary>

![image](https://user-images.githubusercontent.com/6015058/126866066-51e61325-8048-487f-bf81-49599d637e20.png)

After walking a bit:

![image](https://user-images.githubusercontent.com/6015058/126866081-6fb563a0-a4d3-4149-930b-69122d1b99f0.png)

</details>

**Comment by mcpower** [comment]
This seems to be fixed with a newer version (probably from b7e81a222bbfda4ad4ba22589920f1347c201d76 onwards):

- Disabled these FTB mods in the modpack:
  - Banners
  - Chunks
  - Essentials
  - Library
  - Quests
  - Ranks
  - Teams
- Added `ftb-chunks-1605.2.3-build.75` and `ftb-gui-library-1605.2.1.41-forge`, downloaded from CurseForge

Could not reproduce the problem afterwards.

**Comment by mcpower** [comment]
Please disregard the above comment - `ftb-chunks-1605.2.3-build.75` is an *older* version of the mod.

Building directly at HEAD (as of writing this is 30be819fd21b16c4dc3a4a9a25afd91db611b147), replacing `ftb-chunks-forge-1605.3.1-build.24` with the built file at HEAD and updating Architectury to `architectury-1.20.28-forge` fixes the problem - so it *is* indeed fixed in a newer version.
